### ISSUE TITLE
allow access to IMAP servers

### ISSUE THREAD

**Issue by zeigerpuppy** [sandstorm-io/sandstorm#1609]
The current implementation of email apps is very cumbersome, 
It doesn't make much sense to create new email addresses just for sandstorm.

This is confusing to the user because an alias can be mapped but they no longer see their IMAP folders or the emails within them.  

It would be simpler to have the option of the client logging into an IMAP server using the usual credentials.

**Comment by kentonv** [comment]
FWIW, I agree that the current approach to email is more of a toy than something most people can seriously use. We would like to improve this.

Supporting syncing with external IMAP is one thing we want to do. I would like to explore using the Nylas sync server here, as it deals with a lot of historic nastiness for us.

Of course, some people would rather have Sandstorm be the canonical home of their email, without the need for any third party. To that end we need to make Sandstorm be able to operate as an SMTP server for arbitrary domains, the hardest part of which is verifying that the user owns said domain.

Since email is pretty complicated and since there are a lot of existing solutions out there that work, we aren't focusing on it immediately. I expect we'll make a series attempt at email in the second half of 2016.

**Comment by zeigerpuppy** [comment]
I'm not familiar with Nylas, but just had a bit of a read, does seem to be reinventing the wheel a little bit.

In our systems we use a roundcube webclient with a relatively standard postfix/dovecot setup.
it would be nice to be able to integrate these existing accounts into the sandstorm roundcube client.

I agree that authenticating a user to a domain is important but this seems pretty well handled by having the mail client connect to an SMTP server using authentication.  I guess it could be strengthened by also requiring a TLS connection with a valid certificate.  

Email is pretty easily spoofed anyway, but at least in this instance one would know where spam/floods are coming from.

**Comment by paulproteus** [comment]
Hi there!

Short answer is: right now, grain isolation means this can't work, but as
work on the powerbox progresses, we'll be able to make this happen.

**Comment by epiepo** [comment]
Any news on this?

**Comment by kentonv** [comment]
@epiepo No, sorry. E-mail is a big project and while it would certainly be nice to be able to host it on Sandstorm, it's difficult to prioritize over more fundamental features that affect all apps. Though if someone wants to take it on as a contributor, I'll be happy to outline what is needed!
